core/src/test/scala/io/snappydata/SnappyFunSuite.scala

```diff
@@ -40,23 +40,48 @@ abstract class SnappyFunSuite
   protected var testName: String = _
   protected val dirList = ArrayBuffer[String]()
 
+
   protected def sc: SparkContext = {
     val ctx = SnappyContext.globalSparkContext
-    if (ctx != null && !ctx.isStopped) ctx
-    else new SparkContext(newSparkConf())
+    if (ctx != null && !ctx.isStopped) {
+      ctx
+    } else {
+      cachedContext = null
+      new SparkContext(newSparkConf())
+    }
   }
 
   protected def sc(addOn: (SparkConf) => SparkConf): SparkContext = {
     val ctx = SnappyContext.globalSparkContext
-    if (ctx != null && !ctx.isStopped) ctx
-    else new SparkContext(newSparkConf(addOn))
+    if (ctx != null && !ctx.isStopped) {
+      ctx
+    }
+    else {
+      cachedContext = null
+      new SparkContext(newSparkConf(addOn))
+    }
   }
 
   protected def scWithConf(addOn: (SparkConf) => SparkConf): SparkContext = {
     new SparkContext(newSparkConf(addOn))
   }
 
-  protected def snc: SnappyContext = SnappyContext.getOrCreate(sc)
+  private var cachedContext : SnappyContext = null
+
+  def getOrCreate(sc: SparkContext): SnappyContext = {
+    val gnc = cachedContext
+    if (gnc != null) gnc
+    else synchronized {
+      val gnc = cachedContext
+      if (gnc != null) gnc
+      else {
+        cachedContext = SnappyContext(sc)
+        cachedContext
+      }
+    }
+  }
+
+  protected def snc: SnappyContext = getOrCreate(sc)
 
```

Comment:
Does this really require cachedContext and all this, and not be handled by a simple lazy val? Since the lifetime of lazy val is the life of the class, it should be valid for only the tests of one class and then be released when class object goes out of scope (which will happen post afterAll).